### ISSUE TITLE
Do not use comments in reaction bodies as parameters of code generator

### ISSUE THREAD

**Issue by petervdonovan** [lf-lang/lingua-franca#1687]
Let's create an issue for this. The reason behind this that you explained to me in person is _really_ worrisome.

_Originally posted by @lhstrh in https://github.com/lf-lang/lingua-franca/pull/1599#discussion_r1160176312_

 Is it necessary to explain why we should not do this?
* Code that uses this kind of string matching probably does not execute very quickly.
* It relies on keeping a separate enumeration of magic strings.
* It is not idiomatic, so code like this is probably not the easiest to understand.

Arguably, this is a symptom of a larger problem, which is that there is no IR between the parse tree generated by XText and the code generator. As a result, we cannot add our own fields to AST nodes, which makes it tempting to do weird hacks like this.

This is also what we have been doing for unordered reactions as well as for reactions that should not start with initialization code.

**Comment by edwardalee** [comment]
Long ago, I proposed adding a keyword "unordered" for reactions.  I suggested many use cases, including permitting parallel execution.  It was shot down.  Hence this hack.

**Comment by petervdonovan** [comment]
>  It was shot down. Hence this hack.

Yes, it's understandable that this happened given the limitations of the data structures that we are using. My original comment might have come across as a little accusatory, but I had no intention of pointing fingers here since this issue is related to a number of other problems.

**Comment by lhstrh** [comment]
> Long ago, I proposed adding a keyword "unordered" for reactions. I suggested many use cases, including permitting parallel execution. It was shot down. Hence this hack.

@petervdonovan's point is that we don't use the intermediate representation well for storing compile-time information. The reason for this is that we have XText automatically generate our IR from the grammar, implying that any compile-time information we want to store has to correspond to a language feature that is exposed to the user (which led to this whole discussion about the `unordered` keyword). I'm not 100% this implication is true, but, to this day, I'm unaware of a mechanism that let's us modify our Ecore model without getting into the way of Xtext, but I see this limitation as the real problem: _making changes to our IR should not hinge on our willingness to introduce language features_, because that is just silly.

**Comment by cmnrd** [comment]
@petervdonovan makes a great observation! I have noticed these problems too, but never brought them to the point like this. Introducing our own IR (be it a textual syntax or just a model consisting of some record classes) would really help us to both reduce our reliance on XText and modularize our own code base. With the current eCore model it is overly cumbersome to implement transformations, and it seems impossible to add annotations to that model that are not part of the LF syntax. So by introducing an IR (or maybe even multiple IRs at some point) we would have much more control over the information we need and also how the model can be transformed. We can have a small translation layer from the eCore model to the IR, apply transformations, do additional passes where needed and then generate code from there. The only problem is that this will be a ton of work...

**Comment by lsk567** [comment]
I am new to this discussion, but one thing I've explored is to use ANTLR to generate separate parse trees for MTL formulas in #1271. Perhaps this approach could be used to generate separate parse trees for reaction bodies to provide better manipulability.
